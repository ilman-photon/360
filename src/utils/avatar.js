export function stringToColor() {
  return Math.floor(Math.random() * 16777215).toString(16);
}

export function stringAvatar(name = "EyeCare User") {
  const firstCharName = name.split(" ")[0] ? name.split(" ")[0][0] : "";
  const lastCharName = name.split(" ")[1] ? name.split(" ")[1][0] : "";
  return {
    sx: {
      bgcolor: stringToColor(),
    },
    children: `${firstCharName}${lastCharName}`,
  };
}
