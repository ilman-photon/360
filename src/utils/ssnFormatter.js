export function formatSocialSecurity(val) {
  val = val.replace(/\D/g, "");
  val = val.replace(/^(\d{3})/, "***-");
  val = val.replace(/-(\d{3})/, "-***-");
  val = val.replace(/(\d)-(\d{4}).*/, "$1-$2");
  return val;
}
