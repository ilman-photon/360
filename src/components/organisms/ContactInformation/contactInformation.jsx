import PermContactCalendarOutlinedIcon from "@mui/icons-material/PermContactCalendarOutlined";
import {
  Avatar,
  Button,
  Divider,
  Fade,
  Grid,
  Stack,
  Typography,
  useMediaQuery,
} from "@mui/material";
import LabelWithInfo from "../../atoms/LabelWithInfo/labelWithInfo";
import AccountCard from "../../molecules/AccountCard/accountCard";
import styles from "./contactInformation.module.scss";
import { useForm, Controller } from "react-hook-form";
import { StyledInput } from "../../atoms/Input/input";
import { StyledButton } from "../../atoms/Button/button";
import { colors } from "../../../styles/theme";
import EditOutlinedIcon from "@mui/icons-material/EditOutlined";

import { startTransition, Suspense, useCallback, useEffect } from "react";
import Image from "next/image";
import { Regex } from "../../../utils/regex";
import RowRadioButtonsGroup from "../../atoms/RowRadioButtonsGroup/rowRadioButtonsGroup";
import { formatPhoneNumber } from "../../../utils/phoneFormatter";
// import { AddressAutofill } from '@mapbox/search-js-react';
import dynamic from "next/dynamic";

let ClientAddressAutofill;

startTransition(async () => {
  ClientAddressAutofill = await dynamic(
    () => import("@mapbox/search-js-react").then((mod) => mod.AddressAutofill),
    { ssr: false }
  );

  ClientAddressAutofill.accessToken = process.env.MAPBOX_API_TOKEN;
});

export default function ContactInformation({
  autoFillAPIToken = "",
  userData = {},
  isEditing = true,
  OnSaveClicked = () => {
    // This is intended
  },
  OnCancelEditClicked = () => {
    // This is intended
  },
  OnEditClicked = () => {
    // This is intended
  },
}) {
  const { handleSubmit, control, watch, reset } = useForm({
    // defaultValues: DEFAULT_CONTACT_INFO,
    defaultValues: userData, // Object.assign({}, userData),
  });

  const isDesktop = useMediaQuery("(min-width: 769px)");

  const communicationOptions = [
    { label: "Email", value: "email" },
    { label: "Phone", value: "phone" },
    { label: "Both", value: "both" },
  ];

  const [watchedEmail, watchedMobile, watchedPreferredCommunication] = watch([
    "email",
    "mobile",
    "preferredCommunication",
  ]);

  const isOneOfPreferredValid = (name, value) => {
    switch (name) {
      case "email":
        if (watchedPreferredCommunication === "phone") return true;
        else if (watchedPreferredCommunication === "email" && !value)
          return false;
        else if (watchedEmail || watchedMobile) return true;
        break;
      case "phone":
        if (watchedPreferredCommunication === "email") return true;
        else if (watchedPreferredCommunication === "phone" && !value)
          return false;
        else if (watchedEmail || watchedMobile) return true;
        break;
      default:
        return false;
    }
  };

  useEffect(() => {
    if (userData) reset(userData);
  }, [userData]);

  const handleCancel = () => {
    reset(userData);
    OnCancelEditClicked();
  };

  const onSubmit = (data) => {
    console.log({ data });
    OnSaveClicked(data);
  };

  return (
    <AccountCard
      titleIcon={<PermContactCalendarOutlinedIcon />}
      title="Contact Information"
      isEditing={isEditing}
      // OnEditClicked={OnEditClicked}
      actionContent={
        isDesktop ? (
          <Button
            onClick={OnEditClicked}
            variant="text"
            className={styles.editButton}
          >
            <EditOutlinedIcon sx={{ width: 20, height: 20 }} />
            <div type="link" style={{ marginLeft: 4 }}>
              Edit
            </div>
          </Button>
        ) : (
          <StyledButton
            mode="primary"
            size="small"
            onClick={OnEditClicked}
            sx={{ my: 4 }}
          >
            Edit
          </StyledButton>
        )
      }
    >
      <Fade in={!isEditing} unmountOnExit>
        <Stack spacing={3} divider={<Divider />}>
          <LabelWithInfo label="Phone Number">
            {userData.mobile ? formatPhoneNumber(userData.mobile) : ""}
          </LabelWithInfo>

          <LabelWithInfo label="Email ID">
            {userData.email || "-"}
          </LabelWithInfo>

          <LabelWithInfo label="Address">
            {userData.address || "-"}
          </LabelWithInfo>

          <LabelWithInfo label="City">{userData.city || "-"}</LabelWithInfo>

          <Grid container>
            <Grid item xs={6} p={0}>
              <LabelWithInfo label="State">
                {userData.state || "-"}
              </LabelWithInfo>
            </Grid>

            <Grid item xs={6} p={0}>
              <LabelWithInfo label="Zip">{userData.zip || "-"}</LabelWithInfo>
            </Grid>
          </Grid>

          <LabelWithInfo label="Prefered Mode(s) of communication">
            <span style={{ textTransform: "capitalize" }}>
              {userData.preferredCommunication === "both"
                ? "Mobile,Email"
                : userData.preferredCommunication || "-"}
            </span>
          </LabelWithInfo>
        </Stack>
      </Fade>
      <Fade in={isEditing} unmountOnExit>
        <form onSubmit={handleSubmit(onSubmit)}>
          <Stack spacing={3} divider={<Divider />}>
            <Controller
              name="mobile"
              control={control}
              render={({
                field: { onChange, value },
                fieldState: { error },
              }) => {
                return (
                  <StyledInput
                    type="phone"
                    id="mobile"
                    label="Mobile Number"
                    value={value}
                    onChange={onChange}
                    error={!!error}
                    size="small"
                    variant="filled"
                    helperText={error ? error.message : null}
                    xs={{ margin: 0 }}
                  />
                );
              }}
              rules={{
                validate: {
                  required: (value) => {
                    if (!isOneOfPreferredValid("phone", value))
                      return "Email ID or Mobile Number is required";
                  },
                },
                pattern: {
                  value: Regex.isValidPhoneFormat,
                  message: "Incorrect format",
                },
              }}
            />
            <Controller
              name="email"
              control={control}
              render={({
                field: { onChange, value },
                fieldState: { error },
              }) => {
                return (
                  <StyledInput
                    type="text"
                    id="email"
                    label="Email"
                    value={value}
                    onChange={onChange}
                    error={!!error}
                    size="small"
                    variant="filled"
                    helperText={error ? error.message : null}
                  />
                );
              }}
              rules={{
                validate: {
                  required: (value) => {
                    if (!isOneOfPreferredValid("email", value))
                      return "Email ID or Mobile Number is required";
                  },
                },
                pattern: {
                  value: /[a-z0-9]+@[a-z]+\.[a-z]{2,3}/i,
                  message: "Incorrect email format",
                },
              }}
            />

            <Suspense fallback={`Loading...`}>
              <ClientAddressAutofill accessToken={autoFillAPIToken}>
                <Controller
                  name="address"
                  control={control}
                  render={({
                    field: { onChange, value },
                    fieldState: { error },
                  }) => {
                    return (
                      <StyledInput
                        type="text"
                        id="address"
                        label="Address"
                        autoComplete="address-line1"
                        placeholder="Start typing your address, e.g. 123 United States..."
                        sx={{ width: "100%" }}
                        value={value}
                        onChange={onChange}
                        error={!!error}
                        size="small"
                        variant="filled"
                        helperText={error ? error.message : null}
                      />
                    );
                  }}
                  rules={
                    {
                      // required: "This field is required",
                      // pattern: {
                      //   value: Regex.isValidPhoneFormat,
                      //   message: "Incorrect format",
                      // },
                    }
                  }
                />
              </ClientAddressAutofill>
            </Suspense>

            <Controller
              name="city"
              control={control}
              render={({
                field: { onChange, value },
                fieldState: { error },
              }) => {
                return (
                  <StyledInput
                    type="text"
                    id="city"
                    label="City"
                    autoComplete="address-level2"
                    value={value}
                    onChange={onChange}
                    error={!!error}
                    size="small"
                    variant="filled"
                    helperText={error ? error.message : null}
                  />
                );
              }}
              rules={
                {
                  // required: "This field is required",
                  // pattern: {
                  //   value: Regex.isValidPhoneFormat,
                  //   message: "Incorrect format",
                  // },
                }
              }
            />

            <Grid container columnSpacing={2}>
              <Grid item xs={8} style={{ paddingLeft: 0 }}>
                <Controller
                  name="state"
                  control={control}
                  render={({
                    field: { onChange, value },
                    fieldState: { error },
                  }) => {
                    return (
                      <StyledInput
                        type="text"
                        id="state"
                        label="State"
                        autoComplete="address-level1"
                        value={value}
                        onChange={onChange}
                        error={!!error}
                        size="small"
                        variant="filled"
                        helperText={error ? error.message : null}
                        sx={{ width: "100%" }}
                      />
                    );
                  }}
                  rules={
                    {
                      // required: "This field is required",
                      // pattern: {
                      //   value: Regex.isValidPhoneFormat,
                      //   message: "Incorrect format",
                      // },
                    }
                  }
                />
              </Grid>

              <Grid item xs={4} p={0}>
                <Controller
                  name="zip"
                  control={control}
                  render={({
                    field: { onChange, value },
                    fieldState: { error },
                  }) => {
                    return (
                      <StyledInput
                        type="text"
                        id="zip"
                        label="Zip"
                        autoComplete="postal-code"
                        value={value}
                        onChange={onChange}
                        error={!!error}
                        size="small"
                        variant="filled"
                        helperText={error ? error.message : null}
                      />
                    );
                  }}
                  rules={{
                    // required: "This field is required",
                    pattern: {
                      value: /\b\d{5}\b/g,
                      message: "Incorrect format",
                    },
                  }}
                />
              </Grid>
            </Grid>

            <Controller
              name="preferredCommunication"
              control={control}
              render={({
                field: { onChange, value },
                fieldState: { error },
              }) => {
                return (
                  <>
                    <RowRadioButtonsGroup
                      error={!!error}
                      value={value}
                      onChange={onChange}
                      label="Preferred mode of Communication"
                      options={communicationOptions}
                      helperText={error ? error.message : null}
                    />
                  </>
                );
              }}
              rules={{ required: "This field is required" }}
            />

            <Stack direction="row" justifyContent="flex-end" spacing={2}>
              <Button
                onClick={handleCancel}
                variant="contained"
                className={[styles.formButton, styles.outlined].join(" ")}
              >
                Cancel
              </Button>
              <Button
                type="submit"
                variant="contained"
                className={[styles.formButton, styles.primary].join(" ")}
              >
                Save
              </Button>
            </Stack>
          </Stack>
        </form>
      </Fade>
    </AccountCard>
  );
}
