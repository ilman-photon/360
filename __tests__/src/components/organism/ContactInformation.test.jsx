import {
  render,
  fireEvent,
  waitFor,
  within,
  screen,
} from "@testing-library/react";
import "@testing-library/jest-dom";
import ContactInformation from "../../../../src/components/organisms/ContactInformation/contactInformation";
import mediaQuery from "css-mediaquery";
import { formatSocialSecurity } from "../../../../src/utils/ssnFormatter";

window.scrollTo = jest.fn();

function createMatchMedia(width) {
  return (query) => ({
    matches: mediaQuery.match(query, { width }),
    addListener: () => {},
    removeListener: () => {},
  });
}

describe("ContactInformation Components", () => {
  let container;
  const mockUserdata = {
    address: "100 River Road",
    age: 63,
    city: "Johar baru",
    dob: "2022-08-18T13:08:18.012Z",
    email: "Justus4@gmail.com",
    firstName: "Karlie",
    gender: "Transgender",
    issuedCardBack: "https://loremflickr.com/275/173",
    issuedCardFront: "https://loremflickr.com/275/173",
    lastName: "Ernser",
    mobile: "",
    name: "Rupert Jerde",
    preferredCommunication: "both",
    preferredName: "---",
    profilePhoto: {
      name: "my-photo.jpg",
      source: "https://loremflickr.com/640/480",
    },
    ssn: formatSocialSecurity("*****6119"),
    state: "FL",
    title: "Mrs.",
    zip: "24861",
  };
  const mockCallBack = jest.fn();
  beforeEach(async () => {
    container = render(
      <ContactInformation
        isEditing={false}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
  });

  it("PersonalInformation View render", () => {
    expect(container.getByText("Phone Number")).toBeInTheDocument();
    expect(container.getByText("Email Id")).toBeInTheDocument();
    expect(container.getByText("Address")).toBeInTheDocument();
    expect(container.getByText("City")).toBeInTheDocument();
    expect(container.getByText("State")).toBeInTheDocument();
    expect(container.getByText("Zip")).toBeInTheDocument();
    expect(
      container.getByText("Prefered Mode(s) of communication")
    ).toBeInTheDocument();

    expect(container.getByText("Justus4@gmail.com")).toBeInTheDocument();
    expect(container.getByText("100 River Road")).toBeInTheDocument();
    expect(container.getByText("FL")).toBeInTheDocument();
    expect(container.getByText("24861")).toBeInTheDocument();
  });

  it("is email isEmpty with preferred phone", async () => {
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
    const email = container.getByTestId(/email-input-test/i);
    const preferredPhone = container.getByTestId(/phone-test/i);

    fireEvent.change(email, { target: { value: "" } });
    fireEvent.click(preferredPhone);
    await waitFor(() => fireEvent.click(container.getByTestId(/save-button/i)));
  });

  it("is email isEmpty with preferred email", async () => {
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
    const email = container.getByTestId(/email-input-test/i);
    const preferredEmail = container.getByTestId(/email-test/i);

    fireEvent.change(email, { target: { value: "" } });
    fireEvent.click(preferredEmail);
    await waitFor(() => fireEvent.click(container.getByTestId(/save-button/i)));
  });

  it("is email isEmpty with email and mobile is fill", async () => {
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
    const email = container.getByTestId(/email-input-test/i);
    const phone = container.getByTestId(/phone-input-test/i);
    const preferredBoth = container.getByTestId(/both-test/i);

    fireEvent.change(email, { target: { value: "" } });
    fireEvent.change(phone, { target: { value: "09090909090909" } });
    fireEvent.click(preferredBoth);

    await waitFor(() => fireEvent.click(container.getByTestId(/save-button/i)));
  });

  it("is email isEmpty with both selected", async () => {
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
    const email = container.getByTestId(/email-input-test/i);
    const preferredBoth = container.getByTestId(/both-test/i);

    fireEvent.change(email, { target: { value: "" } });
    fireEvent.click(preferredBoth);

    await waitFor(() => fireEvent.click(container.getByTestId(/save-button/i)));
  });

  it("is email isEmpty with both selected", async () => {
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
    const zip = container.getByTestId(/zip-input-test/i);
    fireEvent.change(zip, { target: { value: "12620" } });
    await waitFor(() => fireEvent.click(container.getByTestId(/save-button/i)));
  });

  it("is email isEmpty with both selected", async () => {
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
    const cancelButton = container.getByTestId(/cancel-button/i);
    await waitFor(() => fireEvent.click(cancelButton));
  });

  it("is city incorrect value", async () => {
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
    const cityInput = container.getByTestId(/city-input-test/i);
    fireEvent.change(cityInput, { target: { value: 12620 } });
    await waitFor(() => fireEvent.click(container.getByTestId(/save-button/i)));
  });

  it("is city empty value", async () => {
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
    const cityInput = container.getByTestId(/city-input-test/i);
    fireEvent.change(cityInput, { target: { value: "" } });
    await waitFor(() => fireEvent.click(container.getByTestId(/save-button/i)));
  });

  it("is zip empty value", async () => {
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
    const zipInput = container.getByTestId(/zip-input-test/i);
    fireEvent.keyDown(zipInput, { key: "Enter", code: "Enter", charCode: 13 });
    // await waitFor(() =>
    //   fireEvent.keyDown(container.getByTestId(/save-button/i))
    // );
  });

  it("onChange autocomplete", async () => {
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
    const addressAutocomplete = container.getByTestId(
      /address-autocomplete-test/i
    );
    const input = within(addressAutocomplete).getByRole("combobox");
    addressAutocomplete.focus();

    fireEvent.change(input, { target: { value: "jakarta" } });
    fireEvent.keyDown(addressAutocomplete, { key: "ArrowDown" });
    fireEvent.keyDown(addressAutocomplete, { key: "Enter" });
  });

  it("render with is desktop", async () => {
    window.matchMedia = createMatchMedia("1920px");
    container = render(
      <ContactInformation
        isEditing={true}
        userData={mockUserdata}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
  });

  it("render with is empty user data", async () => {
    window.matchMedia = createMatchMedia("1920px");
    container = render(<ContactInformation isEditing={true} userData={{}} />);
  });

  it("render with mobile phone", async () => {
    window.matchMedia = createMatchMedia("1920px");
    container = render(
      <ContactInformation
        isEditing={true}
        userData={{
          mobile: "0999999999",
          preferredCommunication: "",
        }}
        OnEditClicked={mockCallBack}
        OnCancelEditClicked={mockCallBack}
        OnSaveClicked={mockCallBack}
      />
    );
  });
});
