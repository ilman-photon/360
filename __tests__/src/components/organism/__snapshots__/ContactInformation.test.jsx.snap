// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactInformation Components ContactInformation render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root card  css-1v5z4dq-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-4bhyjd-MuiCardContent-root"
        >
          <div
            class="css-16q27bo-MuiStack-root"
            style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
          >
            <div
              class="css-nen11g-MuiStack-root"
            >
              <div
                class="labelContainer"
              >
                <div
                  aria-label="Phone Number"
                  aria-live="Phone Number"
                  class="label"
                  role="text"
                  tabindex="0"
                >
                  Phone Number
                </div>
              </div>
              <div
                class="css-1okj3ks-MuiStack-root"
              >
                <div
                  style="margin-top: 10px;"
                >
                  <div
                    aria-label="(706) 509-6731"
                    tabindex="0"
                  >
                    (706) 509-6731
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <div
              class="css-nen11g-MuiStack-root"
            >
              <div
                class="labelContainer"
              >
                <div
                  aria-label="Email ID"
                  aria-live="Email ID"
                  class="label"
                  role="text"
                  tabindex="0"
                >
                  Email ID
                </div>
              </div>
              <div
                class="css-1okj3ks-MuiStack-root"
              >
                <div
                  style="margin-top: 10px;"
                >
                  <div
                    aria-label="Justus4@gmail.com"
                    tabindex="0"
                  >
                    Justus4@gmail.com
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <div
              class="css-nen11g-MuiStack-root"
            >
              <div
                class="labelContainer"
              >
                <div
                  aria-label="Address"
                  aria-live="Address"
                  class="label"
                  role="text"
                  tabindex="0"
                >
                  Address
                </div>
              </div>
              <div
                class="css-1okj3ks-MuiStack-root"
              >
                <div
                  style="margin-top: 10px;"
                >
                  <div
                    aria-label="645 Benedict Cliff"
                    tabindex="0"
                  >
                    645 Benedict Cliff
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <div
              class="css-nen11g-MuiStack-root"
            >
              <div
                class="labelContainer"
              >
                <div
                  aria-label="City"
                  aria-live="City"
                  class="label"
                  role="text"
                  tabindex="0"
                >
                  City
                </div>
              </div>
              <div
                class="css-1okj3ks-MuiStack-root"
              >
                <div
                  style="margin-top: 10px;"
                >
                  <div
                    aria-label="Daphneeshire"
                    tabindex="0"
                  >
                    Daphneeshire
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <div
              class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1lojpzt-MuiGrid-root"
              >
                <div
                  class="css-nen11g-MuiStack-root"
                >
                  <div
                    class="labelContainer"
                  >
                    <div
                      aria-label="State"
                      aria-live="State"
                      class="label"
                      role="text"
                      tabindex="0"
                    >
                      State
                    </div>
                  </div>
                  <div
                    class="css-1okj3ks-MuiStack-root"
                  >
                    <div
                      style="margin-top: 10px;"
                    >
                      <div
                        aria-label="South Dakota"
                        tabindex="0"
                      >
                        South Dakota
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1lojpzt-MuiGrid-root"
              >
                <div
                  class="css-nen11g-MuiStack-root"
                >
                  <div
                    class="labelContainer"
                  >
                    <div
                      aria-label="Zip"
                      aria-live="Zip"
                      class="label"
                      role="text"
                      tabindex="0"
                    >
                      Zip
                    </div>
                  </div>
                  <div
                    class="css-1okj3ks-MuiStack-root"
                  >
                    <div
                      style="margin-top: 10px;"
                    >
                      <div
                        aria-label="03245"
                        tabindex="0"
                      >
                        03245
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <hr
              class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
            />
            <div
              class="css-nen11g-MuiStack-root"
            >
              <div
                class="labelContainer"
              >
                <div
                  aria-label="Prefered Mode(s) of communication"
                  aria-live="Prefered Mode(s) of communication"
                  class="label"
                  role="text"
                  tabindex="0"
                >
                  Prefered Mode(s) of communication
                </div>
              </div>
              <div
                class="css-1okj3ks-MuiStack-root"
              >
                <div
                  style="margin-top: 10px;"
                >
                  <span
                    style="text-transform: capitalize;"
                  >
                    <div
                      aria-label="email"
                      tabindex="0"
                    >
                      email
                    </div>
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        style="position: fixed; bottom: 16px; right: 16px;"
      >
        <button
          aria-label="Edit option"
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root customButton customButtonSmall   css-5kb46n-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Edit
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root card  css-1v5z4dq-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root css-4bhyjd-MuiCardContent-root"
      >
        <div
          class="css-16q27bo-MuiStack-root"
          style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
        >
          <div
            class="css-nen11g-MuiStack-root"
          >
            <div
              class="labelContainer"
            >
              <div
                aria-label="Phone Number"
                aria-live="Phone Number"
                class="label"
                role="text"
                tabindex="0"
              >
                Phone Number
              </div>
            </div>
            <div
              class="css-1okj3ks-MuiStack-root"
            >
              <div
                style="margin-top: 10px;"
              >
                <div
                  aria-label="(706) 509-6731"
                  tabindex="0"
                >
                  (706) 509-6731
                </div>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="css-nen11g-MuiStack-root"
          >
            <div
              class="labelContainer"
            >
              <div
                aria-label="Email ID"
                aria-live="Email ID"
                class="label"
                role="text"
                tabindex="0"
              >
                Email ID
              </div>
            </div>
            <div
              class="css-1okj3ks-MuiStack-root"
            >
              <div
                style="margin-top: 10px;"
              >
                <div
                  aria-label="Justus4@gmail.com"
                  tabindex="0"
                >
                  Justus4@gmail.com
                </div>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="css-nen11g-MuiStack-root"
          >
            <div
              class="labelContainer"
            >
              <div
                aria-label="Address"
                aria-live="Address"
                class="label"
                role="text"
                tabindex="0"
              >
                Address
              </div>
            </div>
            <div
              class="css-1okj3ks-MuiStack-root"
            >
              <div
                style="margin-top: 10px;"
              >
                <div
                  aria-label="645 Benedict Cliff"
                  tabindex="0"
                >
                  645 Benedict Cliff
                </div>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="css-nen11g-MuiStack-root"
          >
            <div
              class="labelContainer"
            >
              <div
                aria-label="City"
                aria-live="City"
                class="label"
                role="text"
                tabindex="0"
              >
                City
              </div>
            </div>
            <div
              class="css-1okj3ks-MuiStack-root"
            >
              <div
                style="margin-top: 10px;"
              >
                <div
                  aria-label="Daphneeshire"
                  tabindex="0"
                >
                  Daphneeshire
                </div>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
          >
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1lojpzt-MuiGrid-root"
            >
              <div
                class="css-nen11g-MuiStack-root"
              >
                <div
                  class="labelContainer"
                >
                  <div
                    aria-label="State"
                    aria-live="State"
                    class="label"
                    role="text"
                    tabindex="0"
                  >
                    State
                  </div>
                </div>
                <div
                  class="css-1okj3ks-MuiStack-root"
                >
                  <div
                    style="margin-top: 10px;"
                  >
                    <div
                      aria-label="South Dakota"
                      tabindex="0"
                    >
                      South Dakota
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1lojpzt-MuiGrid-root"
            >
              <div
                class="css-nen11g-MuiStack-root"
              >
                <div
                  class="labelContainer"
                >
                  <div
                    aria-label="Zip"
                    aria-live="Zip"
                    class="label"
                    role="text"
                    tabindex="0"
                  >
                    Zip
                  </div>
                </div>
                <div
                  class="css-1okj3ks-MuiStack-root"
                >
                  <div
                    style="margin-top: 10px;"
                  >
                    <div
                      aria-label="03245"
                      tabindex="0"
                    >
                      03245
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <hr
            class="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
          />
          <div
            class="css-nen11g-MuiStack-root"
          >
            <div
              class="labelContainer"
            >
              <div
                aria-label="Prefered Mode(s) of communication"
                aria-live="Prefered Mode(s) of communication"
                class="label"
                role="text"
                tabindex="0"
              >
                Prefered Mode(s) of communication
              </div>
            </div>
            <div
              class="css-1okj3ks-MuiStack-root"
            >
              <div
                style="margin-top: 10px;"
              >
                <span
                  style="text-transform: capitalize;"
                >
                  <div
                    aria-label="email"
                    tabindex="0"
                  >
                    email
                  </div>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      style="position: fixed; bottom: 16px; right: 16px;"
    >
      <button
        aria-label="Edit option"
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root customButton customButtonSmall   css-5kb46n-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Edit
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
